mobile/bindings.go

```diff
@@ -62,29 +76,37 @@ func Start(extraArgs string, unlockerReady, rpcReady Callback) {
 		return
 	}
 
-	// Set up channels that will be notified when the RPC servers are ready
-	// to accept calls.
+	// Set a channel that will be notified when the RPC server is ready to
+	// accept calls.
 	var (
-		unlockerListening = make(chan struct{})
-		rpcListening      = make(chan struct{})
+		rpcListening = make(chan struct{})
+		quit         = make(chan struct{})
 	)
 
-	// We call the main method with the custom in-memory listeners called
-	// by the mobile APIs, such that the grpc server will use these.
+	// We call the main method with the custom in-memory listener called by
+	// the mobile APIs, such that the grpc server will use it.
 	cfg := lnd.ListenerCfg{
-		WalletUnlocker: &lnd.ListenerWithSignal{
-			Listener: walletUnlockerLis,
-			Ready:    unlockerListening,
-		},
 		RPCListener: &lnd.ListenerWithSignal{
 			Listener: lightningLis,
 			Ready:    rpcListening,
 		},
 	}
 
+	// We only support a single lnd instance at a time (singleton) for now,
+	// so we make sure to return immediately if it has already been
+	// started.
+	if !atomic.CompareAndSwapInt32(&lndStarted, 0, 1) {
```

Comment:
Good catch! Yeah, I meant to remove the first one.... I think it makes sense to flip the bit here, since we'll be sure the defer that flips it back will eventually run.

Reply:
@halseth Actually for the second call if lnd has already been started, I don't think this code will ever be reached, as it will stop with the `signal.Intercept()` at line 54 before it.

Reply:
Ah, because we added a similar atomic var in there?

We could probably move this check up, but make sure we reset the var in case an error happens before we actually start lnd.

Reply:
> Ah, because we added a similar atomic var in there?

Yeah exactly. We have a similar one there.

> We could probably move this check up, but make sure we reset the var in case an error happens before we actually start lnd.

Indeed, It made sense why you wanted to have this one here just before the goroutines, but it seems like it doesn't work out correctly.